src/tools/w3dviewer/w3dviewdoc.cpp

```diff
@@ -1018,23 +1021,181 @@ void CW3DViewDoc::SetFogColor(Vector3 &color)
     }
 }
 
-#if 0
-bool CW3DViewDoc::SaveAggregtate(const char *name)
+PrototypeClass *CW3DViewDoc::GenerateLOD(const char *name, bool format)
 {
-    // do later
-}
+    RenderObjIterator *iter = W3DAssetManager::Get_Instance()->Create_Render_Obj_Iterator();
 
-void CW3DViewDoc::GenerateLOD(const char *name, int type)
-{
-    // do later
+    if (iter == nullptr) {
+        return nullptr;
+    }
+
+    int count = 0;
+    int index = 0xFFFF;
+    char c = 'Z';
+
+    for (iter->First(); !iter->Is_Done(); iter->Next()) {
+        const char *robjname = iter->Current_Item_Name();
+
+        if (W3DAssetManager::Get_Instance()->Render_Obj_Exists(robjname)
+            && iter->Current_Item_Class_ID() == RenderObjClass::CLASSID_HLOD && strstr(robjname, name) == robjname) {
+            const char *str = &robjname[strlen(name)];
+
+            if (format != 0) {
+                if (format == 1) {
+                    if (*str >= 'a' && *str <= 'z' || *str >= 'A' && *str <= 'Z') {
+                        char c1 = str[1];
+
+                        if (!c1) {
+                            count++;
+
+                            if (format) {
+                                char c2 = toupper(robjname[strlen(robjname) - 1]);
+                                if (c >= c2) {
+                                    c = c2;
+                                }
+                            } else {
+                                int i2 = atoi(&robjname[strlen(robjname) - 1]);
+                                if (index >= i2) {
+                                    index = i2;
+                                }
+                            }
+                        }
+                    }
+                }
+            } else if (*str == 'L' || *str == 'l') {
+                if (str[1] >= '0' && str[1] <= '9') {
+                    char c1 = str[2];
+
+                    if (!c1) {
+                        count++;
+
+                        if (format) {
+                            char c2 = toupper(robjname[strlen(robjname) - 1]);
+                            if (c >= c2) {
+                                c = c2;
+                            }
+                        } else {
+                            int i2 = atoi(&robjname[strlen(robjname) - 1]);
+                            if (index >= i2) {
+                                index = i2;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    RenderObjClass **objs = new RenderObjClass *[count];
+    RenderObjClass **objs2 = &objs[count - 1];
+
+    for (int i = 0; i < count; i++) {
+        StringClass name;

```

Comment:
Doesn't this shadow `const char *name` now and break this logic?

Reply:
fixed.